x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/routes/check_and_init_asset_criticality_resources.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Logger } from '@kbn/core/server';
+import { AssetCriticalityDataClient } from '../asset_criticality_data_client';
+import type { SecuritySolutionRequestHandlerContext } from '../../../../types';
+
+// As internal user we check for existence of asset crititcality resources
+// and initialise it if it does not exist
+export const checkAndInitAssetCriticalityResources = async (
```

Comment:
It was written in such a way, that we can in any route just add one line, and it will check for asset criticality index. To receive a `assetCriticalityDataClient` we need to initialise every time in route, or in once in the request context factory, which is probably not the best way.



```
 await checkAndInitAssetCriticalityResources(context, logger);
 ```